tasks/main.yaml

```diff
@@ -1,13 +1,10 @@
 ---
+- name: Add apt-repo for Debian/Ubuntu
+  include: add-apt-repo.yaml
+  when: ansible_pkg_mgr == "apt"
 
-- name: Test distribution
-  assert:
-    that: ansible_os_family == "RedHat"
-
-- name: Install elasticsearch package for RedHat
-  yum:
-    name: "{{ elasticsearch_pkg }}"
-    state: present
+- name: Install elasticsearch package
+  action: "{{ ansible_pkg_mgr }} name={{ elasticsearch_pkg }} state=present"
```

Comment:
Please do not use in-line YAML. Also use of `yum` and `apt` modules is preferred over the `action`.